File: algorithms/3_hard/M/01526_minimum_number_of_increments_on_subarrays_to_form_a_target_array/solution.cpp
class Solution {
public:
    int minNumberOperations(vector<int>& target) {

        const size_t len{target.size()};
        int total{target[0]};

        for (size_t i{1}; i < len; ++i) {
            total += max(0, target[i] - target[i - 1]);
        }

        return total;
    }
};

File: algorithms/3_hard/M/01526_minimum_number_of_increments_on_subarrays_to_form_a_target_array/solution.rs
impl Solution {
    pub fn min_number_operations(target: Vec<i32>) -> i32 {
        target[0] + target.windows(2).into_iter().fold(0_i32, |acc, nums| acc + (nums[1] - nums[0]).max(0_i32))
    }
}
